modules/xmpp/JingleSessionPC.js

```diff
@@ -1457,19 +1598,19 @@ export default class JingleSessionPC extends JingleSession {
      *  the 'doReplaceStream' task or the 'doAddStream' task (for example)
      */
     addLocalTracks(tracks) {
-        const workFunction = () => {
-            if (!this.peerconnection) {
-                return 'Error: tried adding stream with no active peer'
-                    + ' connection';
-            }
-            for (const track of tracks) {
-                this.peerconnection.addTrack(track);
+        const queueTask = new ModificationQueueTask(thisTask => {
+            if (this.peerconnection) {
+                for (const track of tracks) {
+                    this.peerconnection.addTrack(track);
+                    thisTask.setShouldRenegotiate(true);
+                }
+            } else {
+                thisTask.setError(
+                    'Error: tried adding stream with no active peerconnection');
             }
+        });
 
-            return true;
-        };
-
-        return this._doRenegotiate('addStreams', workFunction);
+        return this._doRenegotiate('addStreams', queueTask);
```

Comment:
it's a bit confusing to me to call this method `_doRenegotiate` when we then have a separate variable (set inside the worker function) that is tested inside of `_doRenegotiate` saying whether or not we should renegotiate.  maybe it was misleading before, too; i can't remember.